src/QirRuntime/build-qir-runtime.ps1

```diff
@@ -1,68 +1,94 @@
-    }
+if (($IsMacOS) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Darwin"))))
+{
+    Write-Host "On MacOS build QIR Runtime using the default C/C++ compiler (should be AppleClang)"
+}
+elseif (($IsLinux) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Lin"))))
+{
+    Write-Host "On Linux build QIR Runtime using Clang"
+    $env:CC = "clang-11"
+    $env:CXX = "clang++-11"
+    $env:RC = "clang++-11"
+    $clangTidy = "-DCMAKE_CXX_CLANG_TIDY=clang-tidy-11"
+}
+elseif (($IsWindows) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Win"))))
+{
+    Write-Host "On Windows build QIR Runtime using Clang"
+    $env:CC = "clang.exe"
+    $env:CXX = "clang++.exe"
+    $env:RC = "clang++.exe"
+    $llvmExtras = Join-Path $PSScriptRoot externals LLVM
+    $env:PATH += ";$llvmExtras"
 
-    $qirRuntimeBuildFolder = (Join-Path $PSScriptRoot "build\$Env:BUILD_CONFIGURATION")
-    if (-not (Test-Path $qirRuntimeBuildFolder)) {
-        New-Item -Path $qirRuntimeBuildFolder -ItemType "directory"
+    if (!(Get-Command clang -ErrorAction SilentlyContinue) -and (choco find --idonly -l llvm) -contains "llvm") {
+        # LLVM was installed by Chocolatey, so add the install location to the path.
+        $env:PATH += ";$($env:SystemDrive)\Program Files\LLVM\bin"
     }
 
-    Push-Location $qirRuntimeBuildFolder
+    if (Get-Command clang-tidy -ErrorAction SilentlyContinue) {
+        # Only run clang-tidy if it's installed. This is because the package used by chocolatey on
+        # the build pipeline doesn't include clang-tidy, so we allow skipping that there and let
+        # the Linux build catch tidy issues.
+        $clangTidy = "-DCMAKE_CXX_CLANG_TIDY=clang-tidy"
+    }
+} else {
+    Write-Host "##vso[task.logissue type=error;]Failed to identify the OS. Will use default CXX compiler"
+}
 
-    cmake -G Ninja $clangTidy -D CMAKE_BUILD_TYPE="$Env:BUILD_CONFIGURATION" ../..
-    cmake --build . --target install
+$qirRuntimeBuildFolder = (Join-Path $PSScriptRoot "build\$Env:BUILD_CONFIGURATION")
+if (-not (Test-Path $qirRuntimeBuildFolder)) {
+    New-Item -Path $qirRuntimeBuildFolder -ItemType "directory"
+}
 
-    Pop-Location
+Push-Location $qirRuntimeBuildFolder
 
-    $env:CC = $oldCC
-    $env:CXX = $oldCXX
-    $env:RC = $oldRC
+cmake -G Ninja $clangTidy -D CMAKE_BUILD_TYPE="$Env:BUILD_CONFIGURATION" ../..
+if ($LastExitCode -ne 0) {
+    Write-Host "##vso[task.logissue type=error;]Failed to generate QIR Runtime."
+}
+cmake --build . --target install
+if ($LastExitCode -ne 0) {
+    Write-Host "##vso[task.logissue type=error;]Failed to build QIR Runtime."
+}
 
-    if ($LastExitCode -ne 0) {
-        Write-Host "##vso[task.logissue type=error;]Failed to build QIR Runtime."
-    }
-} else {
-    Write-Host "To enable build of QIR Runtime set ENABLE_QIRRUNTIME environment variable to 'true'"
+$os = "win32"
+$pattern = "*.dll"
+if ($IsMacOS) {
```

Comment:
Out of curiosity:
Why _32_ (in "win32")? Why not just "win"?

Reply:
You know, to be honest, it's just a bad habit leftover from working on Windows... I type it without meaning to and don't even register it on the screen when I read it! There is no reason for the "32" here and it's probably more of a distraction than anything. I'll update my next PR to remove this!

Reply:
Fix it in https://github.com/microsoft/qsharp-runtime/pull/553/commits/065589a3492580c0682ff40ef31eb873558fb581